src/UI Component/Login/signuppage.js

```diff
@@ -0,0 +1,194 @@
+import React, { useState, useEffect } from "react";
+import SignIn from "../Login/login"
+
+
+const SignUpPage = () => {
+  const [formData, setFormData] = useState({
+    firstname: "",
+    lastname: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+  });
+
+  const [errors, setErrors] = useState({});
+  const [isSignUpSuccess, setIsSignUpSuccess] = useState(false);
+  const [showLogin, setShowLogin] = useState(false);
+
+  useEffect(() => {
+    if (isSignUpSuccess) {
+      // Clear form fields upon successful sign-up
+      setFormData({
+        firstname: "",
+        lastname: "",
+        email: "",
+        password: "",
+        confirmPassword: "",
+      });
+    }
+  }, [isSignUpSuccess]);
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setFormData({
+      ...formData,
+      [name]: value,
+    });
+  };
+
+  const handleToggleForm = () => {
+    setShowLogin(!showLogin);
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    // Perform validation
+    const validationErrors = validateForm(formData);
+    setErrors(validationErrors);
+
+    // If there are no validation errors, you can submit the form
+    if (Object.keys(validationErrors).length === 0) {
+      // Alert successful sign-up
+      alert("Successful Sign Up!");
+
+      // Set the sign-up success status
+      setIsSignUpSuccess(true);
+    }
+  };
+
+  const validateForm = (data) => {
+    let errors = {};
+
+    if (!data.firstname.trim()) {
+      errors.firstname = "Firstname is required";
+    }
+    if (!data.lastname.trim()) {
+      errors.lastname = "Lastname is required";
+    }
+
+    if (!data.email.trim()) {
+      errors.email = "Email is required";
+    } else if (!/\S+@\S+\.\S+/.test(data.email)) {
```

Comment:
Please explain what this ?

Reply:
Also, this is a nice approach to handle form errors but please implement this using [Yup](https://www.npmjs.com/package/yup)

Reply:
So, the entire regular expression /^\S+@\S+\.\S+$/ is checking if the data.email string has the basic structure of an email address. The ! in front negates the result, meaning that the condition is true if the email does not match the specified pattern.  More like  it assesses whether the email is not in the anticipated format. When the email is not in the expected structure, the validation error associated with the email is shown.